--- a/autoload/wheel/pendulum.vim
+++ b/autoload/wheel/pendulum.vim
@@ -4,11 +4,12 @@
 
 fun! wheel#pendulum#record ()
 	" Add current torus, circle, location to history
+	let history = g:wheel_history
 	let [torus, circle, location] = wheel#referen#location('all')
 	let coordin = [torus.name, circle.name, location.name]
 	if index(g:wheel_history, coordin) < 0
 		let g:wheel_history = insert(g:wheel_history, coordin, 0)
 	endif
 	let max = g:wheel_config.max_history
 	let g:wheel_history = g:wheel_history[:max - 1]
 endfu
